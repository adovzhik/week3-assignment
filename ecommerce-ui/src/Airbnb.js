import React from 'react';
import PropTypes from 'prop-types';
import Card from 'react-bootstrap/Card';
import Button from 'react-bootstrap/Button';

class Airbnb extends React.Component {

  static propTypes = {
    title: PropTypes.string.isRequired,
    houseType: PropTypes.string,
    image: PropTypes.string,
    city: PropTypes.string,
    country: PropTypes.string,
    cost: PropTypes.number.isRequired,
    onClick: PropTypes.func.isRequired,
    description: PropTypes.string,
    isInCart: PropTypes.bool,
    buttonText: PropTypes.string.isRequired,
    buttonVarient: PropTypes.string,
    idx: PropTypes.number
  }

  render() {
    const {
      title, houseType, image, city, country, cost, onClick, 
      description, isInCart, buttonText, buttonVarient, idx
    } = this.props;

    return (
      <Card style={isInCart ? { width: '12rem' } : null}>
        <Card.Img src={image} />
        <Card.Body>
          <Card.Title>{title}</Card.Title>
          <Card.Text>
            <small>{city}, {country} â€¢ {houseType}</small>
            <br />
            ${cost} per night
            <br />
            <i>{description}</i>
          </Card.Text>
        </Card.Body>
        <Card.Footer>
          <Button
            variant={buttonVarient}
            onClick={() => onClick(idx)}>
            {buttonText}
          </Button>
        </Card.Footer>
      </Card>
    );
  }
}

export default Airbnb;
