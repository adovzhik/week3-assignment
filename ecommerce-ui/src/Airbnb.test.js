import React from 'react';
import Airbnb from './Airbnb';
import { shallow } from 'enzyme';

it('shallow renders', () => {
  const wrapper = shallow(
    <Airbnb
      title={"test property"}
      houseType={"house"}
      image={"https://images.unsplash.com/photo-1475855581690-80accde3ae2b?ixlib=rb-1.2.1&auto=format&fit=crop&w=1650&q=80"}
      city={"Portland"}
      country={"USA"}
      cost={100}
      onClick={() => console.log("clicked!")}
      description={"test description"}
      isInCart={false}
      buttonText={"Add"}
      buttonVarient={"primary"}
      idx={0}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
