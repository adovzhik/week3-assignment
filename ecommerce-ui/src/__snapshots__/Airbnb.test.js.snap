// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`shallow renders 1`] = `
<Card
  as="div"
  body={false}
  style={null}
>
  <CardImg
    as="img"
    src="https://images.unsplash.com/photo-1475855581690-80accde3ae2b?ixlib=rb-1.2.1&auto=format&fit=crop&w=1650&q=80"
    variant={null}
  />
  <CardBody>
    <CardTitle>
      test property
    </CardTitle>
    <CardText>
      <small>
        Portland
        , 
        USA
         â€¢ 
        house
      </small>
      <br />
      $
      100
       per night
      <br />
      <i>
        test description
      </i>
    </CardText>
  </CardBody>
  <CardFooter>
    <Button
      active={false}
      disabled={false}
      onClick={[Function]}
      type="button"
      variant="primary"
    >
      Add
    </Button>
  </CardFooter>
</Card>
`;
